pkg/lib/cockpit-components-modifications.jsx

```diff
@@ -130,67 +122,61 @@
  * Pass string `shell` and `ansible` with scripts.
  *
  */
-export class Modifications extends React.Component {
-    constructor(props) {
-        super(props);
-        this.state = {
-            showDialog: false,
-        };
-    }
+export const Modifications = ({ entries, failed, permitted, title, shell, ansible }) => {
+    const [showDialog, setShowDialog] = React.useState(false);
 
-    render() {
-        let emptyRow = null;
-        let fail_message = this.props.permitted ? _("No system modifications") : _("The logged in user is not permitted to view system modifications");
-        fail_message = this.props.failed ? _("Error running semanage to discover system modifications") : fail_message;
-        if (this.props.entries === null) {
-            emptyRow = <DataListItem>
-                <DataListItemRow>
-                    <DataListItemCells dataListCells={[<DataListCell key="loading">{_("Loading system modifications...")}</DataListCell>]} />
-                </DataListItemRow>
-            </DataListItem>;
-        }
-        if (this.props.entries?.length === 0) {
-            emptyRow = <DataListItem>
-                <DataListItemRow>
-                    <DataListItemCells dataListCells={[<DataListCell key={fail_message}>{fail_message}</DataListCell>]} />
-                </DataListItemRow>
-            </DataListItem>;
-        }
+    let emptyRow = null;
+    let fail_message = permitted ? _("No system modifications") : _("The logged in user is not permitted to view system modifications");
+    fail_message = failed || fail_message;
+    if (entries === null) {
+        emptyRow = <DataListItem>
+            <DataListItemRow>
+                <DataListItemCells dataListCells={[<DataListCell key="loading">{_("Loading system modifications...")}</DataListCell>]} />
+            </DataListItemRow>
+        </DataListItem>;
+    }
+    if (entries?.length === 0) {
+        emptyRow = <DataListItem>
+            <DataListItemRow>
+                <DataListItemCells dataListCells={[<DataListCell key={fail_message}>{fail_message}</DataListCell>]} />
+            </DataListItemRow>
+        </DataListItem>;
```

Comment:
These 6 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19370-20230922-100423-663bbd9c-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)